### PR TITLE
Feat/vault secrets operator

### PR DESCRIPTION
**Pull request by ghost**
This installs the new beta release of the secrets operator in a dedicated namespace.

### CODE REVIEW

File: Provisioning/Vault/terraform.tfvars [hunk 1]
```diff
@@ -0,0 +1,2 @@
+kubecontext = "kind-kind"
```
Comment (in0rdr) [inline Provisioning/Vault/terraform.tfvars:1]: This file should not even exist. Have a look at https://github.com/adfinis/Vault-Playground-on-Kubernetes/pull/7

File: Provisioning/Vault/vault-secrets-operator.tf [hunk 2]
```diff
@@ -0,0 +1,70 @@
+resource "vault_auth_backend" "kubernetes" {
+  type = "kubernetes"
+}
+
+# Use local service account token and CA cert
+# https://developer.hashicorp.com/vault/docs/auth/kubernetes
+resource "vault_kubernetes_auth_backend_config" "kubernetes" {
+  backend            = vault_auth_backend.kubernetes.path
+  kubernetes_host    = "https://kubernetes.default.svc"
+  issuer             = "https://kubernetes.default.svc.cluster.local"
```
Comment (in0rdr) [inline Provisioning/Vault/vault-secrets-operator.tf:10]: - [ ] Todo: disable_iss_validation=true is the new recommended value for all versions of Vault (https://developer.hashicorp.com/vault/docs/auth/kubernetes#discovering-the-service-account-issuer)

**Review by Xelef2000 — DISMISSED** [review]

File: Provisioning/Vault/terraform.tfvars (hunk 1 above)
Comment (in0rdr) [reply Provisioning/Vault/terraform.tfvars]: replaced it with an example tfvars file

**Comment by in0rdr** [comment]
* Did a rebase on the main branch to incorporate all the formatting changes from the pre-commit hooks
* Added a Markdown file with basic documentation in VAULT-SECRETS-OPERATOR.md

@Xelef2000 ready for review

**Review by Xelef2000 — APPROVED** [review]

**Pull request merged**